app/Connect/Actions/SubmitLeaderboardAction.php

```diff
@@ -0,0 +1,242 @@
+{
+    use GeneratesLegacyAuditComment;
+
+    protected int $leaderboardId;
+    protected int $gameId;
+    protected int $achievementSetId;
+    protected string $title;
+    protected string $description;
+    protected string $startTrigger;
+    protected string $submitTrigger;
+    protected string $cancelTrigger;
+    protected string $valueDefinition;
+    protected bool $lowerIsBetter;
+    protected string $format;
+
+    public function execute(User $user,
+        ?int $leaderboardId, ?int $gameId, ?int $achievementSetId,
+        string $title, string $description,
+        string $startTrigger, string $submitTrigger, string $cancelTrigger,
+        string $valueDefinition, bool $lowerIsBetter, string $format): array
+    {
+        if (!$leaderboardId && !$gameId && !$achievementSetId) {
+            return $this->missingParameters();
+        }
+
+        $this->leaderboardId = $leaderboardId ?? 0;
+        $this->gameId = $gameId ?? 0;
+        $this->achievementSetId = $achievementSetId ?? 0;
+        $this->title = $title;
+        $this->description = $description;
+        $this->startTrigger = $startTrigger;
+        $this->submitTrigger = $submitTrigger;
+        $this->cancelTrigger = $cancelTrigger;
+        $this->valueDefinition = $valueDefinition;
+        $this->lowerIsBetter = $lowerIsBetter;
+        $this->format = $format;
+
+        return $this->process();
+    }
+
+    protected function initialize(Request $request): ?array
+    {
+        if (!$request->has('i')) { // if an existing leaderboard ID is not provided
+            if (!$request->has('g') && !$request->has('p')) { // either game or set is required
+                return $this->missingParameters();
+            }
+        }
+
+        // all properties must be provided regardless of update/create
+        if (!$request->has(['n', 'd', 's', 'b', 'c', 'l', 'w', 'f', 'h'])) {
+            return $this->missingParameters();
+        }
+
+        $this->leaderboardId = request()->integer('i', 0);
+        $this->gameId = request()->integer('g', 0);
+        $this->achievementSetId = request()->integer('p', 0);
+
+        $this->title = request()->input('n') ?? '';
+        $this->description = request()->input('d') ?? '';
+        $this->startTrigger = request()->input('s') ?? '';
+        $this->submitTrigger = request()->input('b') ?? '';
+        $this->cancelTrigger = request()->input('c') ?? '';
+        $this->valueDefinition = request()->input('l') ?? '';
+        $this->lowerIsBetter = request()->boolean('w', false);
+        $this->format = request()->input('f') ?? '';
+
+        $checksum = request()->input('h') ?? '';
+        if (!$this->checksumMatches($checksum, $this->user->display_name)) {
+            if ($this->user->User === $this->user->display_name || !$this->checksumMatches($checksum, $this->user->User)) {
+                return $this->accessDenied('Invalid checksum.');
+            }
+        }
```

Comment:
I've enabled the checksum validation in this PR. We've been sending it basically forever.